src/plugins/expressions/common/expression_functions/specs/map_column.ts

```diff
@@ -44,7 +44,7 @@ export const mapColumn: ExpressionFunctionDefinition<
       types: ['string', 'null'],
       help: i18n.translate('expressions.functions.mapColumn.args.idHelpText', {
         defaultMessage:
-          'An optional id of the resulting column. When `null` the name/column argument is used as id.',
+          'An optional id of the resulting column. When no id is provided, the id will be looked up from the existing column and default to the name.',
```

Comment:
I had trouble inferring everything that happens from this description. Maybe like this?  `... will be looked up from the existing column by the provided name argument. If no column with this name exists yet, a new column with this name and an identical id will be added to the table`